src/mono/mono/metadata/loaded-images.c

```diff
@@ -100,15 +101,26 @@ mono_loaded_images_remove_image (MonoImage *image)
 
 	loaded_images         = mono_loaded_images_get_hash (li, image->ref_only);
 	loaded_images_by_name = mono_loaded_images_get_by_name_hash (li, image->ref_only);
-	image2 = (MonoImage *)g_hash_table_lookup (loaded_images, image->name);
+
+	char *name = image->name;
```

Comment:
This is actually invalid C++ because of the goto on line 97. It's one of the most common things you'll run into when trying to write C++-compatible C, and you can see it in the mono/mono build. Error:
```
loaded-images.c:97:3: error: cannot jump from this goto statement to its label
                goto done;
                ^
loaded-images.c:105:8: note: jump bypasses variable initialization
        char *name = image->name;
              ^
loaded-images.c:90:3: error: cannot jump from this goto statement to its label
                goto done;
                ^
loaded-images.c:105:8: note: jump bypasses variable initialization
        char *name = image->name;
              ^
2 errors generated.
```